src/darwin/Framework/CHIPTests/MTRCertificateTests.m

```diff
@@ -504,7 +532,13 @@ - (void)testGenerateCSR
     __auto_type * publicKey = [MTRCertificates publicKeyFromCSR:csr error:nil];
     XCTAssertNotNil(publicKey);
 
-    SecKeyRef originalKeyRef = [testKeys publicKey];
+    SecKeyRef originalKeyRef;
+    if ([testKeys respondsToSelector:@selector(copyPublicKey)]) {
+        originalKeyRef = [testKeys copyPublicKey];
```

Comment:
This also looks like a leak.

Reply:
It was!

Reply:
(added `CFAutorelease(originalKeyRef);` below - it has been a while, scrutiny welcome)

Reply:
(specifically:  is manual creation of `NSAutoreleasePool` or use of `@autoreleasepool` merited here?

Reply:
No.

Reply:
You also need to nil check CFAutorelease, please just use [object autorelease] per our chat

Reply:
(Otherwise you have a crash here potentially)

Reply:
(discussing)